### ISSUE TITLE
Media tech for testing

### ISSUE THREAD

**Issue by carpasse** [videojs/video.js#1981]
I have been doing some unit testing on the video player and from time to time I encounter problems because whenever we add the player to the page it tries to download the unexisting video or complains that the video is not there or not supported.

I want to propose the creation of a "test tech" that will work with sources of extension '.test' and that will mimic the html5 tech but without doing the actual requests.

What do you think?

**Comment by heff** [comment]
Yeah, this would be great to have for external tests. We have a [mediafaker](https://github.com/videojs/video.js/blob/master/test/unit/mediafaker.js) that we've been using for unit tests. It gets used via the [test helpers makePlayer function](https://github.com/videojs/video.js/blob/master/test/unit/test-helpers.js#L17).

It sounds like it'd be good to extract this into it's own thing that could be used by other projects as well. If you'd be interested in helping with that, that'd be great.

We're right in the middle of  major refactor, so if you start anything soon you should start off @mmcc's [es6 branch](https://github.com/videojs/video.js/pull/1976).

**Comment by carpasse** [comment]
I don't think I will be able to start working on it this week. 
Do you know when is @mmcc going to be merged to the master??

**Comment by heff** [comment]
Just giving some of the other contributors time to review it before merging in, but I would guess later this week.

**Comment by gkatsev** [comment]
es6 branch has been merged :P
It's also been released. Now is the time!
